transport/src/actors/gateway.rs

```diff
@@ -127,6 +128,14 @@ impl GatewayBehaviour {
         Some(GatewayEvent::Ping { peer_id, ping })
     }
 
+    fn get_query_id(&mut self, req_id: &OutboundRequestId) -> Option<String> {
```

Comment:
Maybe call it `take_query_id`? It's unexpected for the `get` method to mutate `self`

Reply:
Sure.

Reply:
Amended.